planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp

```diff
@@ -820,54 +705,57 @@ PathWithLaneId PullOverModule::generateStopPath() const
   return stop_path;
 }
 
-bool PullOverModule::planShiftPath(const Pose goal_pose)
+void PullOverModule::incrementPathIndex()
 {
-  std::vector<ShiftParkingPath> valid_paths;
+  status_.current_path_idx =
+    std::min(status_.current_path_idx + 1, status_.pull_over_path.partial_paths.size() - 1);
+}
 
-  const auto & route_handler = planner_data_->route_handler;
-  const auto & current_pose = planner_data_->self_pose->pose;
-  const auto & common_parameters = planner_data_->parameters;
+PathWithLaneId PullOverModule::getCurrentPath() const
+{
+  return status_.pull_over_path.partial_paths.at(status_.current_path_idx);
+}
 
-  if (!isLongEnough(status_.current_lanes, goal_pose) || status_.pull_over_lanes.empty()) {
-    return false;
+PathWithLaneId PullOverModule::getFullPath() const
+{
+  PathWithLaneId path{};
+  const auto & paths = status_.pull_over_path.partial_paths;
+  for (size_t i = 0; i < paths.size(); ++i) {
+    if (i == 0) {
+      path.points.insert(path.points.end(), paths.at(i).points.begin(), paths.at(i).points.end());
+    } else {
+      // skip overlapping point
+      path.points.insert(
+        path.points.end(), next(paths.at(i).points.begin()), paths.at(i).points.end());
+    }
   }
 
-  // find candidate paths
-  const auto pull_over_paths = pull_over_utils::generateShiftParkingPaths(
-    *route_handler, status_.current_lanes, status_.pull_over_lanes, current_pose, goal_pose,
-    common_parameters, parameters_);
-  if (pull_over_paths.empty()) {
-    return false;
-  }
+  return util::removeOverlappingPoints(path);
+}
+
+bool PullOverModule::planShiftPath(const Pose goal_pose)
```

Comment:
reference for goal_pose

Reply:
this function is no longer necessary, so I will remove it.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/1911/commits/b47d458c9870cfcb74d01d9866159bd273ee1972